### PR TITLE
Bump plugin to Java 11

### PR DESCRIPTION
**Pull request by mkemmerz**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Doings

- Bump the plugin to java 11
- Use Java 11 nodes to build the plugin in CI/CD
- Bump dependencies to mirror newest versions (e.g. jenkinci-plugins, junit and some other util dependencies)
- Bump property version for jenkins.version to 2.372
- Set working directory of surefire plugin to ${java.io.tmpdir}
- Introduce @Rule for testcases, those are failing otherwise with java 11

### CODE REVIEW

**Comment by mkemmerz** [comment]
Almost done. Weirdly enough I have to create a tmp folder manually in the /target for the tests to run correctly. The Rule for the temporary folder seems to have issues here? Maybe related to my Windows system?

**Comment by StefanSpieker** [comment]
That looks quite promissing. I think we can improve it a little bit, please take a look at #34 - you can remove a cuple of versions and pick up the numbers from the parent.

**Comment by mkemmerz** [comment]
I merged https://github.com/jenkinsci/thin-backup-plugin/pull/34 into this branch so it is based on your pom updates now.

File: src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/BackupDirStructureSetup.java [hunk 1]
```diff
@@ -30,9 +33,7 @@ public class BackupDirStructureSetup {
 
   @Before
   public void setup() throws Exception {
-    final File tempDir = new File(System.getProperty("java.io.tmpdir"));
-    backupDir = new File(tempDir, "BackupDirForHudsonBackupTest");
-    backupDir.mkdir();
+    backupDir = tmpFolder.getRoot();
```
Comment (StefanSpieker) [inline src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/BackupDirStructureSetup.java:36]:
```suggestion
    backupDir = tmpFolder.newFolder("thin-backup");
```

File: src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupMatrixJob.java [hunk 2]
```diff
@@ -1,47 +1,41 @@
...
   private File jenkinsHome;
 
   @Before
   public void setup() throws IOException, InterruptedException {
-    File base = new File(System.getProperty("java.io.tmpdir"));
-    backupDir = TestHelper.createBackupFolder(base);
+    backupDir = TestHelper.createBackupFolder(tmpFolder.getRoot());
 
-    jenkinsHome = TestHelper.createBasicFolderStructure(base);
+    jenkinsHome = TestHelper.createBasicFolderStructure(tmpFolder.getRoot());
```
Comment (StefanSpieker) [inline src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupMatrixJob.java:32]: see above.

File: src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupSet.java [hunk 3]
```diff
@@ -102,7 +102,7 @@ public void testBackupSetContainsDirectory() throws IOException {
 
     final File tempDir = new File(System.getProperty("java.io.tmpdir"));
     backupDir = new File(tempDir, "BackupDirForHudsonBackupTest");
-    final File testFile = new File(backupDir, "tempFile.nxt");
+    final File testFile = new File(tmpFolder.getRoot(), "tempFile.nxt");
```
Comment (StefanSpieker) [inline src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupSet.java:105]:
```suggestion
    final File testFile = tmpFolder.newFile("tempFile.nxt");
```

**Review by StefanSpieker — CHANGES_REQUESTED** [review]

File: src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupMatrixJob.java [hunk 4]
```diff
@@ -1,47 +1,41 @@
...
+  public TemporaryFolder tmpFolder = new TemporaryFolder(new File(System.getProperty("java.io.tmpdir")));
+
   private File backupDir;
   private File jenkinsHome;
 
   @Before
   public void setup() throws IOException, InterruptedException {
-    File base = new File(System.getProperty("java.io.tmpdir"));
-    backupDir = TestHelper.createBackupFolder(base);
+    backupDir = TestHelper.createBackupFolder(tmpFolder.getRoot());
```
Comment (StefanSpieker) [inline src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupMatrixJob.java:30]:
```suggestion
    backupDir = TestHelper.createBackupFolder(tmpFolder.newFolder("thin-backup"));

```

File: src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupMatrixJob.java [hunk 5]
```diff
@@ -1,47 +1,42 @@
...
 
   @Before
   public void setup() throws IOException, InterruptedException {
-    File base = new File(System.getProperty("java.io.tmpdir"));
-    backupDir = TestHelper.createBackupFolder(base);
+    backupDir = TestHelper.createBackupFolder(tmpFolder.newFolder("thin-backup"));
+
 
-    jenkinsHome = TestHelper.createBasicFolderStructure(base);
+    jenkinsHome = TestHelper.createBasicFolderStructure(tmpFolder.getRoot());
```
Comment (StefanSpieker) [inline src/test/java/org/jvnet/hudson/plugins/thinbackup/backup/TestBackupMatrixJob.java:33]:
```suggestion
    jenkinsHome = TestHelper.createBasicFolderStructure(tmpFolder.newFolder("jenkins"));
```

**Review by StefanSpieker — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by StefanSpieker** [comment]
Thanks a lot @mkemmerz for that. The next release will include your changes.

**Comment by mkemmerz** [comment]
Let me know if you have some other implementation ideas for other plugins @StefanSpieker 😁

**Pull request merged**